### PR TITLE
include: ipc4: copier: clean it from unrelated structs

### PR DESCRIPTION
**Pull request by ujfalusi**
Hi,

for some reason the copier.h contains the definition of `struct ipc4_base_module_cfg_ext` which it should not have as it belongs to base-config.h 
Clean this up and remove an unused enum as well from the copier.h.

### CODE REVIEW

File: src/include/ipc4/base-config.h [hunk 1]
```diff
@@ -144,4 +144,19 @@ struct ipc4_output_pin_format {
...
+	/* specifies number of items in output_pins array. Maximum size is 8 */
+	uint16_t nb_output_pins;
+	uint8_t reserved[12];
+	/* Specifies format of input pins. Pin format arrays may be non-continuous i.e.
+	 * may contain pin #0 format followed by pin #2 format in case pin #1 will not be
+	 * in use. FW assigned format of the pin based on pin_index, not on a position of
+	 * the item in the array. Applies to both input and output pins.
+	 */
+	struct ipc4_input_pin_format input_pins[1];
+	struct ipc4_output_pin_format output_pins[1];
```
Comment (ujfalusi) [inline src/include/ipc4/base-config.h:159]: This part looks just awful in my view, would it make sense to replace it with
```c
uint8_t pin_formats[];
```
The `output_pins` in itself can not be referenced at all, if nb_input_pins != 1, it is not pointing where it is expected it to be pointing.

**Comment by ujfalusi** [comment]
Changes since v1:
- to keep the `__attribute__((packed, aligned(4)))` fro ipc4_base_module_cfg_ext use 'git commit -n'

File: src/include/ipc4/base-config.h (hunk 1 above)
Comment (ranj063) [reply src/include/ipc4/base-config.h:159]: makes sense. Can you please fix this in this PR too @ujfalusi

**Comment by ujfalusi** [comment]
Changes since v2:
- replace the input_pins[1] / output_pins[1] from the base_cfg_ext with pin_formats[] and extend the comment to make it clear that we have the input pins followed by output pins formats behind of pin_formats[]
- Add helper to calculate base_cfg_ext size based on number of pins
- add a composite struct to combine base_cfg and base_cfg_ext for easier parsing and casting by modules.

The smart amp test module is updated to use this header, but I don't want to have unrelated patches in this PR. I will create a new PR for that later on.

File: src/include/ipc4/copier.h [hunk 2]
```diff
@@ -70,16 +70,6 @@ struct ipc4_base_module_cfg_ext {
...
-/* This enum defines short 16bit parameters common for all modules.
- * Value of module specific parameters have to be less than 0x3000.
- */
-enum ipc4_base_module_params {
-	/* handled inside LargeConfigGet of module instance */
-	IPC4_MOD_INST_PROPS  = 0xFE,
-	/* handled inside ConfigSet of module instance */
-	IPC4_MOD_INST_ENABLE = 0x3000
-};
-
```
Comment (kv2019i) [inline src/include/ipc4/copier.h:82]: @mwasko @abonislawski @aborisovich @marcinszkudlinski Any plans to use these, or do we go ahead and remove?

**Review by kv2019i — APPROVED** [review]

**Review by ranj063 — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

File: src/include/ipc4/copier.h (hunk 2 above)
Comment (pblaszko) [reply src/include/ipc4/copier.h:82]: Support for MOD_INST_ENABLE is required for IADK module adapter API.
Support for MOD_INST_PROPS is required for both: Module API and IADK module adapter API.

It seems in SOF this implementation is missing but it is required and used at least by Win driver.
I think it should be moved to generic module header, but not removed.

File: src/include/ipc4/copier.h [hunk 3]
```diff
@@ -70,16 +70,6 @@ struct ipc4_base_module_cfg_ext {
 	struct ipc4_output_pin_format output_pins[1];
 } __attribute__((packed, aligned(4)));
 
-/* This enum defines short 16bit parameters common for all modules.
- * Value of module specific parameters have to be less than 0x3000.
- */
-enum ipc4_base_module_params {
-	/* handled inside LargeConfigGet of module instance */
-	IPC4_MOD_INST_PROPS  = 0xFE,
```
Comment (pblaszko) [inline src/include/ipc4/copier.h:78]: Please note we have below: `struct ipc4_module_instance_props`
It is binded to IPC4_MOD_INST_PROPS definition. It should be moved together.

File: src/include/ipc4/copier.h (hunk 2 above)
Comment (pblaszko) [reply src/include/ipc4/copier.h:82]: @wszypelt do we test MOD_INST_PROPS for closed-source ACE FW? do we test it for SOF? I can see the test name is check_mod_inst_props_get() in fw_02_base_fw_api/test_00_debug_api

**Review by pblaszko — CHANGES_REQUESTED** [review]
"include: ipc4: copier: Move and update ipc4_base_module_cfg_ext to base-config.h" - commit title too long?

**Comment by ujfalusi** [comment]
> "include: ipc4: copier: Move and update ipc4_base_module_cfg_ext to base-config.h" - commit title too long?

True, but which part can be dropped to keep the meaning?
` include: ipc4: copier: Move ipc4_base_module_cfg_ext to base-config.h`
`include: ipc4: copier: Move and update ipc4_base_module_cfg_ext`
?

File: src/include/ipc4/copier.h (hunk 2 above)
Comment (pblaszko) [reply src/include/ipc4/copier.h:82]: how about ipc4/base-config.h, the same as you do with ipc4_base_module_cfg_ext ?

**Comment by pblaszko** [comment]
> > "include: ipc4: copier: Move and update ipc4_base_module_cfg_ext to base-config.h" - commit title too long?
> 
> True, but which part can be dropped to keep the meaning? `include: ipc4: copier: Move ipc4_base_module_cfg_ext to base-config.h` `include: ipc4: copier: Move and update ipc4_base_module_cfg_ext` ?

ok, that is not blocker for me :)

File: src/include/ipc4/copier.h (hunk 2 above)
Comment (ujfalusi) [reply src/include/ipc4/copier.h:82]: OK, then I might just use one patch to do these.

**Comment by ujfalusi** [comment]
Changes since v3:
- one patch
- move whole lot of other structs as they do not belong to the copier.h as suggested by @pblaszko

**Review by pblaszko — APPROVED** [review]

File: src/include/ipc4/copier.h (hunk 2 above)
Comment (pblaszko) [reply src/include/ipc4/copier.h:82]: @ujfalusi Thank you!

**Comment by ranj063** [comment]
@ujfalusi there's a failure in the DMIC test in the internal CI. I dont think it is related to this PR but not sure if this is a recent regression

**Comment by lgirdwood** [comment]
@wszypelt @lrudyX not expecting a failure as this is a code move, could someone check. Thanks !

**Comment by ujfalusi** [comment]
> @ujfalusi there's a failure in the DMIC test in the internal CI. I dont think it is related to this PR but not sure if this is a recent regression

@ranj063, none of the moved headers are in use at currently, I don't think it has anything to do with this patch.

**Comment by ujfalusi** [comment]
Changes since v4:
- rebased on main to clear and re-run the tests. DMIC failing should not have anything to do with this PR, it is just moving unused structs...

**Comment by kv2019i** [comment]
@wszypelt Intel CI test is failing and I can't decipher the cause, all tests in logs have passed. This PR only moved header definitions around, so no failures are really expected (and builds seem fine).

**Comment by wszypelt** [comment]
@kv2019i We have problems with one of our rvp today, I repeated the tests, now it should pass

**Comment by ranj063** [comment]
> @kv2019i @lgirdwood We have problems with one of our rvp today, I repeated the tests, now it should pass

@wszypelt it is still failing

**Comment by wszypelt** [comment]
@ranj063 @ujfalusi @kv2019i @lgirdwood We still have a problem with the rvp(dmic), I'm trying to fix it as soon as I can

**Comment by wszypelt** [comment]
@ranj063 @ujfalusi @kv2019i @lgirdwood Unfortunately, the equipment failed, we managed to fix it, I hope that now everything will work, within 30 minutes the tests should be done

**Comment by wszypelt** [comment]
> @ranj063 @ujfalusi @kv2019i @lgirdwood we managed to fix it, I hope that now everything will work, within 30 minutes the tests should be done

all green

**Review by dnikodem — APPROVED** [review]

**Pull request merged**